pages/k8s/certs-and-trust.md

```diff
@@ -0,0 +1,247 @@
+
+## Managing certificates
+
+Unfortunately, the **Juju** controller does not provide any mechanisms for
+generating or distributing additional certificates to be used by the
+applications on the machines, so they must be managed by the charms via the
+secure relation data channel. This is done using the [tls-certificates][]
+[interface protocol][interface] and a relation to an application providing a
+Certificate Authority.  (This CA could be either a root CA, or an intermediary
+CA authorised by some other root CA to issue certificates.)
+
+When the relation is established, the root CA's certificate is sent via the
+relation and [installed][install_ca_cert] as trusted. Then, certificate
+requests can be issued over the relation and new certificates created by the CA
+and returned over the relation.
+
+Because all units with a relation to the CA have a chain of trust up to it (or
+its root CA), they will automatically trust a certificate generated by the CA
+without requiring anything to be communicated from the unit which holds the
+certificate. On the other hand, for the certificates to be trusted externally
+(such as by clients) or by applications without a relation to the CA, the CA
+will have to be configured as an intermediary CA with a chain of trust up to a
+globally trusted CA (such as [Comodo][] or [DigiCert][]).
+
+The certificate information will also be included in the **Kubernetes** `config`
+file that **CDK** generates to be used by [kubelet][], so that communications
+between the local machine and Kubernetes will be secured.
+
+## Server certificates
+
+Each service that will be connected to will need a [server certificate][]
+identifying and validating it to any clients that wish to connect.
+
+The primary address at which the service can be reached will be its [common
+name][CN]; ideally, this will be a fully-qualified domain name (FQDN) which
+will not change, but for internal service communication, it is often just the
+[ingress address][network primitives] for the unit. Any additional names or
+addresses by which the service can be reached will be its [subject alternative
+names (SANs)][SANs].
+
+**CDK** will manage the server certificates automatically, including
+generating the certificate with the proper CN and SANs. However, the
+[kubernetes-master][] charm also supports an [`extra_sans`][] option which
+can be used to provide additional names to be added to the SANs list.
+
+## Client certificates
+
+In order to provide for two-way security, some services require that clients
+identify themselves via a [client certificate][]. These are more or less the
+same as server certificates, but are presented by a client to the service
+they are connecting to, so that the service can validate that client's identity.
+Client certificates can _only_ be used to identify a client and will be
+rejected by clients if presented by a service they are connecting to.
+
+# Certificate Authorities for CDK
```

Comment:
From here this seems more like a practical, actionable page that should probably be in operations.